apps/back-end/src/routes.ts

```diff
@@ -0,0 +1,189 @@
+import { TsRestResponseError } from "@ts-rest/core";
+import { RouterImplementation } from "@ts-rest/fastify";
+import { contract } from "@mykomap/common";
+import { FastifyPluginOptions, FastifyReply, FastifyRequest } from "fastify";
+import fs from "node:fs";
+import path from "node:path";
+
+/** Provides the shared configuration options for the Mykomap router implementation. */
+export interface MykomapRouterConfig extends FastifyPluginOptions {
+  /** Options sepcifically used by Mykomap Api plugin go in here */
+  mykomap: {
+    /** Defines the path to the data store */
+    dataRoot: string;
+  };
+}
+
+//////////////////////////////////////////////////////////////////////
+// Helper types
+
+/** The Mykomap API contract type */
+type Contract = typeof contract;
+
+//////////////////////////////////////////////////////////////////////
+// Helper functions
+
+// Helper functions
+
+/** A predicate testing for URL-safe base64 string (RFC4648 sect 5) */
+function validBase64(value: string): boolean {
+  return !!value.match(/^[A-Z0-9_-]+$/i);
```

Comment:
If I understand the docs correctly, ts-rest should automatically do runtime validation of the request and return 400 if it fails, so we don't need to do it separately
https://ts-rest.com/docs/fastify#request-validation-error-handling

The Zod schema in `contract.ts` just needs the datasetID and filters to be strings, so we maybe need to tighten up these constraints in the schema, then we can get rid of these functions.

Reply:
Main reason for not doing that is I just don't currently know how to make Zod perform arbitrary validations on strings.

Reply:
There's a `z.string().regex()` validation and also a built-in `z.string().base64()`

Reply:
I've addressed this general point in some fixup commits (labelled with `fixup`) later in the chain: all this sort of validation which can be is now done via the contract.

Let me know if you want me to leave those separate before you review them, because I intend to merge the fixup commits with the commit they are modifying before merging the PR. You might prefer to review after I've done that?

Reply:
Looks good, feel free to merge the commits however makes sense